offchain/metadata/fetchers/metadata_fetcher.py

```diff
@@ -57,35 +66,42 @@ def set_timeout(self, timeout: int):  # type: ignore[no-untyped-def]
         """
         self.timeout = timeout
 
+    def _get_async_adapter_for_uri(self, uri: str) -> Optional[Adapter]:
+        if self.async_adapter_configs is not None:
+            for async_adapter_config in self.async_adapter_configs:
```

Comment:
you think this could be slightly rewritten to avoid being deeply nested? also why do we need a for loop with this?